src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2716,12 +2716,16 @@ rct_window* WindowRideConstructionOpen()
         return nullptr;
     }
 
-    if (currentRide->type == RIDE_TYPE_MAZE)
+    const auto& rtd = currentRide->GetRideTypeDescriptor();
+    switch (rtd.ConstructionWindowContext)
     {
-        return context_open_window_view(WV_MAZE_CONSTRUCTION);
+        case RideConstructionWindowContext::Maze:
+            return context_open_window_view(WV_MAZE_CONSTRUCTION);
+        case RideConstructionWindowContext::Default:
+        default:
```

Comment:
There are only two enumerators here, and you list them all, so I would remove the `default`. If someone adds a new one, then the compilation will fail and point out them to fill this up with the new case.
```suggestion
        case RideConstructionWindowContext::Default:
```

Reply:
I had to add a default return it was complaining about the return paths of the function...